packages/block-library/src/site-logo/index.php

```diff
@@ -71,56 +71,83 @@ function register_block_core_site_logo() {
 			'render_callback' => 'render_block_core_site_logo',
 		)
 	);
-	add_filter( 'pre_set_theme_mod_custom_logo', 'sync_site_logo_to_theme_mod' );
-	add_filter( 'theme_mod_custom_logo', 'override_custom_logo_theme_mod' );
 }
 add_action( 'init', 'register_block_core_site_logo' );
 
 /**
- * Overrides the custom logo with a site logo, if the option is set.
- *
- * @param string $custom_logo The custom logo set by a theme.
- *
- * @return string The site logo if set.
+ * Expose the custom_logo theme-mod in the settings REST API.
  */
-function override_custom_logo_theme_mod( $custom_logo ) {
-	$site_logo = get_option( 'site_logo' );
-	return false === $site_logo ? $custom_logo : $site_logo;
-}
+register_setting(
```

Comment:
Registers the theme-mods setting properly and exposes it to the REST API. We're only going to expose the `custom_logo` setting here, but in the future we may also think about adding widgets since they are theme-mods too. Not something related to this PR though